pkg/domain/workflow.go

```diff
@@ -14,16 +14,32 @@ type WorkflowStore interface {
 	Add(ctx context.Context, r *workflow.Workflow) (*workflow.Workflow, error)
 }
 
+// WorkflowAssignmentStore is an inteface to workflow assignment stores
+type WorkflowAssignmentStore interface {
+	Find(context context.Context, workflowName string) *WorkflowAssignment
+	GetAll(context context.Context, workflowName string) []*WorkflowAssignment
+	Add(context context.Context, workflowName *WorkflowAssignment) (*WorkflowAssignment, error)
+}
+
 // WorkflowBackend is the interface for the FuseML workflows
 type WorkflowBackend interface {
-	CreateListener(context.Context, *log.Logger, string, bool) (string, error)
+	CreateListener(context.Context, *log.Logger, string, bool) (*workflow.WorkflowListener, error)
 	CreateWorkflow(context.Context, *log.Logger, *workflow.Workflow) error
 	CreateWorkflowRun(context.Context, string, Codeset) error
 	ListWorkflowRuns(context.Context, workflow.Workflow, WorkflowRunFilter) ([]*workflow.WorkflowRun, error)
+	GetWorkflowListener(context.Context, *log.Logger, string) (*workflow.WorkflowListener, error)
```

Comment:
This feels inconsistent to have _CreateListener_ and _GetWorkflowListener_. Either have _Workflow_ in both names or in none

Reply:
Indeed, I have updated the name of the other methods.